### DISCUSSION TITLE
Delegating openapi file bundeling to an external API

### DISCUSSION THREAD

**Discussion by ghost** [microcks/microcks#1022] [Ideas]
Microcks has a lot of code used for parsing and bundeling openapi files that is out of date with the spec.
This parsing problem has already been solved in [swagger-parser](https://github.com/APIDevTools/swagger-parser). It even has a dereferencer.

Could this tool be integrated into microcks to handle the bundling and the dereferencing of the openapi spec?

Example impl:
- make an http api that downloads the git repository from url and returns the bundled version of the openapi file

- [In Microcks] on reception of a new mock. Call the api with the url of the file in the repository. On reception of the bundled openapi spec continue as usual

**Comment by lbroudoux** [comment]
This is nice for solving our dereferencing/reference navigation issues at the moment, but I see different caveats:
* This is a Javascript/Typescript library and integrating would require an external component with complicated communication or runtime evaluation,
* We have a very lightweight parser in Microcks, and the goal is clearly not to be fully spec-compliant. We prefer to be lightweight, in control of the dependencies we're pulling and fully focused on the problems we're solving.
Please let us a few days/weeks spending time on #986 and #984 to see if we can sort this out.

**Reply by ghost** [reply]
There is also [swagger-parser](https://github.com/swagger-api/swagger-parser) a java library that does swagger parsing but last time i tested this library i couldn't dereference some refs.
